internal/delete_handler.go

```diff
@@ -294,3 +294,40 @@ func getGarbageFromPrefix(folders []storage.Folder, nonGarbage []BackupTime) []s
 	}
 	return garbage
 }
+
+func DeleteBeforeTarget(
+	folder storage.Folder,
+	target storage.Object,
```

Comment:
Let's take as a parameter one variable predicate - object in a store instead of a target and before function.

Reply:
Done